src/lints/inherent_associated_pub_const_missing.ron

```diff
@@ -0,0 +1,86 @@
+    human_readable_name: "inherent impl's associated pub const removed",
+    description: "An inherent impl's associated public const removed or renamed",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#item-remove"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        inherent_impl {
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                            
+                            associated_constant {
+                                associated_constant: name @output @tag
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+
+                                span_: span @optional {
+                                    filename @output
+                                    begin_line @output
+                                }
+                            }    
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @output
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        inherent_impl @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]){
+                            associated_constant {
+                                name @filter(op: "=", value: ["%associated_constant"])
+                            }
+                        }
+                    }
+                }
+                or:
+                    item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @output
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        impl @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]){
+                            implemented_trait {
+                                trait {
+                                    associated_constant {
+                                        name @filter(op: "=", value: ["%associated_constant"])
+                                    }
```

Comment:
Block to check if there is an implemented trait with the const

Reply:
You could actually take this `impl` block and move it into the other `item` block, side by side with its existing `inherent_impl` block.

The query engine will still optimize this query to essentially the same thing as written, but it's still unnecessarily long and confusing for human readers so probably a good idea to apply this change.

Also, a nitpick: try to polish up the small details, like indentation and formatting. Right now, the query has some unexpected indentation, inconsistency in spacing (e.g. no space between `){`), and other elements that seem possibly accidental. While no individual thing like this is a big deal on its own, as a whole they do get in the way and make maintainability harder over time. Sometimes they even cause bugs, and even if they don't, they still make code harder to read and understand for maintainers and new contributors.